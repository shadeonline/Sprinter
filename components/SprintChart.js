import React from 'react';
import { View, Text } from 'react-native';
import { PieChart } from 'react-native-chart-kit';

const SprintChart = ({ totalTasks, backlogTasks, inProgressTasks, completedTasks }) => {
  const chartData = [
    {
      name: 'Backlog',
      population: backlogTasks.length,
      color: '#c3c3c3',
      legendFontColor: 'white',
      legendFontSize: 15,
    },
    {
      name: 'In Progress',
      population: inProgressTasks.length,
      color: '#5386e4',
      legendFontColor: 'white',
      legendFontSize: 15,
    },
    {
      name: 'Completed',
      population: completedTasks.length,
      color: 'green',
      legendFontColor: 'white',
      legendFontSize: 15,
    },
  ];

  const chartConfig = {
    backgroundGradientFrom: '#4C4B63',
    backgroundGradientTo: '#4C4B63',
    color: (opacity = 1) => `rgba(255, 255, 255, ${opacity})`,
  };

  return (
    <View>
      <PieChart
        data={chartData}
        width={300}
        height={220}
        chartConfig={chartConfig}
        accessor="population"
        backgroundColor="transparent"
        paddingLeft="15"
        absolute
      />
      <Text style={{ color: 'white', textAlign: 'center', marginTop: 10 }}>
        Total Tasks: {totalTasks}
      </Text>
    </View>
  );
};

export default SprintChart;
