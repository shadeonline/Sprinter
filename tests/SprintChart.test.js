// SprintChart.test.js

import React from 'react';
import renderer from 'react-test-renderer';
import SprintChart from '../components/SprintChart'; // Adjust the path based on your project structure

// Mock React Navigation
jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: jest.fn(),
    }),
  };
});

// Mock react-native-chart-kit
jest.mock('react-native-chart-kit', () => ({
  PieChart: 'View',
}));

describe('SprintChart', () => {
  it('matches snapshot', () => {
    const totalTasks = 10;
    const backlogTasks = [/* Array of tasks */];
    const inProgressTasks = [/* Array of tasks */];
    const completedTasks = [/* Array of tasks */];

    const tree = renderer.create(
      <SprintChart
        totalTasks={totalTasks}
        backlogTasks={backlogTasks}
        inProgressTasks={inProgressTasks}
        completedTasks={completedTasks}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
