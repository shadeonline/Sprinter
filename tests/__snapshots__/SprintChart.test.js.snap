// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SprintChart matches snapshot 1`] = `
<View>
  <View
    absolute={true}
    accessor="population"
    backgroundColor="transparent"
    chartConfig={
      {
        "backgroundGradientFrom": "#4C4B63",
        "backgroundGradientTo": "#4C4B63",
        "color": [Function],
      }
    }
    data={
      [
        {
          "color": "#c3c3c3",
          "legendFontColor": "white",
          "legendFontSize": 15,
          "name": "Backlog",
          "population": 0,
        },
        {
          "color": "#5386e4",
          "legendFontColor": "white",
          "legendFontSize": 15,
          "name": "In Progress",
          "population": 0,
        },
        {
          "color": "green",
          "legendFontColor": "white",
          "legendFontSize": 15,
          "name": "Completed",
          "population": 0,
        },
      ]
    }
    height={220}
    paddingLeft="15"
    width={300}
  />
  <Text
    style={
      {
        "color": "white",
        "marginTop": 10,
        "textAlign": "center",
      }
    }
  >
    Total Tasks: 
    10
  </Text>
</View>
`;
